exporters/otlp/otlpgrpc/driver.go

```diff
@@ -145,10 +145,13 @@ func (md *metricsDriver) uploadMetrics(ctx context.Context, protoMetrics []*metr
 		if md.metricsClient == nil {
 			return errNoClient
 		}
-		_, err := md.metricsClient.Export(ctx, &colmetricpb.ExportMetricsServiceRequest{
-			ResourceMetrics: protoMetrics,
-		})
-		return err
+		req := func(ctx context.Context) error {
+			_, err := md.metricsClient.Export(ctx, &colmetricpb.ExportMetricsServiceRequest{
+				ResourceMetrics: protoMetrics,
+			})
+			return err
+		}
+		return doRequest(ctx, req, md.connection.cfg.RetrySettings, md.connection.stopCh)
```

Comment:
Should the `doRequest` func be a method on the `connection` type instead of passing state from the type?

Reply:
Sure, I started doing it in this way thinking about how `otlphttp` would use it, but I guess it's easier to refactor if we do plan to use this code there.